packages/dds/tree/src/test/shared-tree/undo.spec.ts

```diff
@@ -453,6 +464,137 @@ describe("Undo and redo", () => {
+			redoStack: redoStack1,
+			unsubscribe: unsubscribe1,
+		} = createTestUndoRedoStacks(originalView.events);
+
+		assert(originalView.root.child !== undefined);
+		originalView.root.child.propertyOne = 256; // 128 -> 256
+
+		const forkedView = originalView.fork();
+
+		const propertyOneUndo = undoStack1.pop();
+		const clonedPropertyOneUndo = propertyOneUndo?.clone(forkedView);
+
+		propertyOneUndo?.revert();
+
+		assert.equal(originalView.root.child?.propertyOne, 128);
+		assert.equal(forkedView.root.child?.propertyOne, 256);
+		assert.equal(propertyOneUndo?.status, RevertibleStatus.Disposed);
+		assert.equal(clonedPropertyOneUndo?.status, RevertibleStatus.Valid);
+
+		clonedPropertyOneUndo?.revert();
+
+		assert.equal(forkedView.root.child?.propertyOne, 128);
+
+		assert.equal(clonedPropertyOneUndo?.status, RevertibleStatus.Disposed);
+	});
+
+	it("reverts original & forked revertibles after making separate changes to the original & forked view", () => {
+		const originalView = asTreeViewAlpha(createLocalSharedTree("testSharedTree"));
+
+		const {
+			undoStack: undoStack1,
+			redoStack: redoStack1,
+			unsubscribe: unsubscribe1,
+		} = createTestUndoRedoStacks(originalView.events);
+
+		assert(originalView.root.child !== undefined);
+		originalView.root.child.propertyOne = 256; // 128 -> 256
+		originalView.root.child.propertyTwo.itemOne = "newItem";
+
+		const forkedView = originalView.fork();
+		const {
+			undoStack: undoStack2,
+			redoStack: redoStack2,
+			unsubscribe: unsubscribe2,
+		} = createTestUndoRedoStacks(forkedView.events);
+
+		assert(forkedView.root.child !== undefined);
+		forkedView.root.child.propertyOne = 512; // 256 -> 512
+
+		undoStack2.pop()?.revert();
+		assert.equal(forkedView.root.child?.propertyOne, 256);
+
+		const undoOriginalPropertyTwo = undoStack1.pop();
+		const clonedUndoOriginalPropertyTwo = undoOriginalPropertyTwo?.clone(forkedView);
+
+		const undoOriginalPropertyOne = undoStack1.pop();
+		const clonedUndoOriginalPropertyOne = undoOriginalPropertyOne?.clone(forkedView);
+
+		undoOriginalPropertyOne?.revert();
+		undoOriginalPropertyTwo?.revert();
+
+		assert.equal(originalView.root.child?.propertyOne, 128);
+		assert.equal(originalView.root.child?.propertyTwo.itemOne, "");
+		assert.equal(forkedView.root.child?.propertyOne, 256);
+		assert.equal(forkedView.root.child?.propertyTwo.itemOne, "newItem");
+
+		clonedUndoOriginalPropertyOne?.revert();
+		clonedUndoOriginalPropertyTwo?.revert();
+
+		assert.equal(forkedView.root.child?.propertyOne, 128);
+		assert.equal(forkedView.root.child?.propertyTwo.itemOne, "");
+
+		assert.equal(undoOriginalPropertyOne?.status, RevertibleStatus.Disposed);
+		assert.equal(undoOriginalPropertyTwo?.status, RevertibleStatus.Disposed);
+		assert.equal(clonedUndoOriginalPropertyOne?.status, RevertibleStatus.Disposed);
+		assert.equal(clonedUndoOriginalPropertyTwo?.status, RevertibleStatus.Disposed);
+	});
+
+	it("reverts cloned revertible on original view", () => {
```

Comment:
you should add a test that expects clone to throw if the view doesn't have the revision (also in the case where the view already reverted something)

Reply:
I think `clone revertible fails if trees are different` already tests "no revision in view" scenario? 

The test tries to `clone(viewB).revert()` on a revertible created from `viewA`. Since, `viewA` and `viewB` but fails since the `revertible` passed to the `viewB` doesn't exist in its `revertibleCommitBranch`. 

+ 
Added `view already reverted something` scenario. Thanks.